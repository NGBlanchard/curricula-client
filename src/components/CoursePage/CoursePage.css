.CoursePage__comment-list {
  list-style-type: none;
  padding: 0;
}

.CoursePage__comment {
  position: relative;
}

.CoursePage__comment::before {
  border-color: transparent #f5f5f5;
  border-style: solid;
  border-width: 0px 0px 10px 10px;
  content: '';
  display: block;
  height: 0px;
  position: absolute;
  width: 0;
  bottom: 14px;
  right: 7px;
}

.CoursePage__comment-text {
  background: #f5f5f5;
  padding: 8px;
  margin-bottom: 4px;
  border-radius: 4px;
}

.CoursePage__comment-user {
  color: #666;
  margin: 8px 0 0;
  font-size: 10pt;
  font-weight: bold;
  text-align: right;
}

#topic {
  font-weight: lighter;
  text-align: center;
}

.comment-card {
  background: none;
  /* box-shadow: 0 4px 5px 0 rgba(202, 80, 255, 0.19); */
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2); 
  border: 1px solid #3f455038;
  border-radius: 4px;
  margin-bottom: 10px;
  font-size: small;
  display: block;
  padding: 10px 10px 0;
  text-align: left;
  width: 100%;
  text-decoration: none;
}

.duration {
  display: flex;
  flex-direction: row;
  
}

.CoursePage__duration {
  position: relative;
  top: 6px;
  left: 15px;
}

.card {
  border-radius: 4px;
  border: 1px solid #3f455038;
  box-shadow: 0 4px 8px 0 rgba(202, 80, 255, 0.19);
  padding: 10px;
  margin-bottom: 20px;
}