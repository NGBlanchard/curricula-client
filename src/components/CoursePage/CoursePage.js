import React, { Component } from 'react'
import CourseContext from '../../context/CourseContext'
import CourseApiService from '../../services/course-api-service'
import { NiceDate, Bull, Section } from '../Utils/Utils'
import CommentForm from '../CommentForm/CommentForm'
import './CoursePage.css'

export default class CoursePage extends Component {
  static defaultProps = {
    match: { params: {} },
  }
  static contextType = CourseContext

  componentDidMount() {
    const { courseId } = this.props.match.params
    this.context.clearError()
    CourseApiService.getCourse(courseId)
      .then(this.context.setCourse)
      .catch(this.context.setError)
    CourseApiService.getCommentsForCourse(courseId)
      .then(this.context.setComments)
      .catch(this.context.setError)
  }

  componentWillUnmount() {
    this.context.clearCourse()
  }

  renderCourse() {
    const { course, comments } = this.context
    return <>
      <h1>{course.title}</h1>
      <h3 id="topic">Topic: {course.topic}</h3>
      <p>
        {/* <CourseStyle course={course} /> */}
        {course.author.id && <>
          <Bull />
          {/* <CourseAuthor course={course.author} /> */}
        </>}
        Posted on
        <Bull />
        <NiceDate date={course.date_created} />
        <Bull />
        by {course.author}
        
      </p>
      <div className="card">
        <h3>Course Description</h3>
        <p className='CoursePage__description'>
          {course.description}
        </p>
      </div>
      <div className="card">
        <h3>Teaching Notes</h3>
        <p className='CoursePage__notes'>
          {course.notes}
        </p>
      </div>
      <div className="card">
        <h3>Readings & Resources</h3>
        <p className='CoursePage__readings'>
          {course.readings}
        </p>
      </div>
      <div className="duration">
        <h3>Estimated Duration </h3>
        <p className='CoursePage__duration'>
          {course.duration} weeks
        </p>
        </div>
        <div>
          <h3>Blurbs</h3>
            <ul className='CoursePage__comment-list'>
              {comments.map((comment, index) =>
                <li key={index} className='comment-card'>
                  <p className='CoursePage__comment-content'>
                    {comment.content}
                  </p>
                </li>
              )}
            </ul>
        </div>
      <CommentForm render={comments}/>
    </>
  }

  render() {
    const { error, course } = this.context
    let content
    if (error) {
      content = (error.error === `Course doesn't exist`)
        ? <p className='red'>Course not found</p>
        : <p className='red'>There was an annoying error</p>
    } else if (!course.id) {
      content = <div className='loading' />
    } else {
      content = this.renderCourse()
    }
    return (
      <Section className='CoursePage'>
        {content}
      </Section>
    )
  }
}


