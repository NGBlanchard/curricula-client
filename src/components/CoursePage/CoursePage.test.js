import CoursePage from './CoursePage'
import { BrowserRouter as Router } from "react-router-dom"
import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() });


describe('<CoursePage />', () => {
  it('Renders without crashing', () => {
    const course = {
        id: 4,
        title: 'title',
        description: 'description',
        readings: 'readings',
        notes: 'notes',
        topic: 'topic',
        date_created: 'date',
        author_id: 2
      }

    const context = {
      clearError(){
        return;
      }, 
      setCourseList(){
        return;
      },
      filterTopics() {
        const filteredtopics = this.state.courseList.map(t => t.topic)
        this.setState({
          topics: filteredtopics
        })
      },
      courseList: [],
      filteredList: [],
    }

    mount(
      <Router>
        <CoursePage
          key={course.id}
          course={course}
        />
      </Router>, context)
  })
})
