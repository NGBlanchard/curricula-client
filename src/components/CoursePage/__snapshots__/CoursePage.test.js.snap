// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoursePage /> renders the page by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CoursePage
      course={
        Object {
          "author_id": 2,
          "date_created": "date",
          "description": "description",
          "id": 4,
          "notes": "notes",
          "readings": "readings",
          "title": "title",
          "topic": "topic",
        }
      }
      key="4"
      match={
        Object {
          "params": Object {},
        }
      }
    >
      <Section
        className="CoursePage"
      >
        <section
          className="Section CoursePage"
        >
          <div
            className="loading"
          />
        </section>
      </Section>
    </CoursePage>
  </Router>
</BrowserRouter>
`;
